openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -235,47 +211,76 @@ def get_cohort(user, course_key, assign=True):
         if not assign:
             return None
 
-    choices = course.auto_cohort_groups
-    if len(choices) > 0:
-        # Randomly choose one of the auto_cohort_groups, creating it if needed.
-        group_name = local_random().choice(choices)
+    course_cohorts = get_course_cohorts(course, assignment_type=CourseUserGroup.RANDOM)
+    if course_cohorts:
+        course_cohort = local_random().choice(course_cohorts)
     else:
-        # Use the "default cohort".
-        group_name = DEFAULT_COHORT_NAME
+        course_cohort, __ = CourseUserGroup.objects.get_or_create(
+            course_id=course_key,
+            group_type=CourseUserGroup.COHORT,
+            name=DEFAULT_COHORT_NAME,
+            defaults={'assignment_type': CourseUserGroup.RANDOM}
+        )
 
-    group, __ = CourseUserGroup.objects.get_or_create(
-        course_id=course_key,
-        group_type=CourseUserGroup.COHORT,
-        name=group_name
+    user.course_groups.add(course_cohort)
+
+    return course_cohort
+
+
+def migrate_cohort_settings(course):
```

Comment:
a few questions about this:
* what made introducing this necessary?
* what is the expected behavior if the values in the modulestore are changed after this function was run?
* i assume that during a course export that values copied in the SQL db will not be taken, but rather the ones originally defined on the course - is this desired?  (cc @explorerleslie @catong )